akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/voice/UserJoinedVoiceConfEvtMsgHdlr.scala

```diff
@@ -13,17 +17,10 @@ trait UserJoinedVoiceConfEvtMsgHdlr extends SystemConfiguration {
   val outGW: OutMsgRouter
 
   def handleUserJoinedVoiceConfEvtMsg(msg: UserJoinedVoiceConfEvtMsg): Unit = {
-    log.info("Received user joined voice conference " + msg)
 
-    if (VoiceUsers.isCallerBanned(msg.body.callerIdNum, liveMeeting.voiceUsers)) {
-      log.info("Ejecting banned voice user " + msg)
-      val event = MsgBuilder.buildEjectUserFromVoiceConfSysMsg(
-        props.meetingProp.intId,
-        props.voiceProp.voiceConf,
-        msg.body.voiceUserId
-      )
-      outGW.send(event)
-    } else {
+    val guestPolicy = GuestsWaiting.getGuestPolicy(liveMeeting.guestsWaiting)
+    val currentTime = System.currentTimeMillis()

```

Comment:
You are only using currentTime once and actually much later. Please consider calling System.... directly there or define ~ in the same block